mysqldump: Add examples to backup/restore all databases with file redirection (#3177)
tldr-pages/tldr: pages/common/mysqldump.md

- Restore a backup, user will be prompted for a password:

`mysql -u {{user}} --password -e "source {{filename.sql}}" {{database_name}}`

> - Backup all databases with compression, user will be prompted for a password:

Reviewer: I think we can leave out the compression part. It is an external command and is not relevant to this page. And after that, please mention the "redirect to a file" part.

Author: Do you mean to exclude the piping to gzip part in the command too? I think it's a common use case that you want to compress your backup files and because `mysqldump` doesn't provide that option by default that an example of it is warranted.

Reviewer: Yes, but that is a separate command. We usually leave that up to the users and just keep the pages focused on a single command.

Author: OK, I've removed the compression part and updated the PR

> - Backup all databases and redirect it to a file, user will be prompted for a password:

Reviewer 2: ```suggestion
- Backup all databases redirecting the output to a file (user will be prompted for a password):
```

Sounds smoother, doesn't it?

> `mysqldump -u {{user}} -p --all-databases > {{filename.sql.gz}}`

Reviewer: Remove .gz

Author: Fixed

Reviewer: I still don't see it in the latest commit - https://github.com/tldr-pages/tldr/pull/3177/files#diff-d2e51c131025b22a22fe2f70f93895a3R16

Author: Sorry, nice catch! It's here https://github.com/pengux/tldr/commit/6e17cbf5e286bba86d3504b97f1c6a2e410901fb

> - Restore a backup with redirection from a file, user will be prompted for a password:

Reviewer 2: ```suggestion
- Restore all databases from a backup (user will be prompted for a password):
```
Sounds clearer, what do you guys think?
